### PR TITLE
Add AWSSigv4Signing

### PR DESCRIPTION
**Pull request by yadavcbala**
*Issue #, if available:*

PR for partial changes - raising to avoid delays during my OOTO

* Using jetpack.io for aws-request-signing-apache-interceptor because there is no maven published jar - refer https://github.com/awslabs/aws-request-signing-apache-interceptor/issues/2
* There is no simple Junit as it requires credentials and works fine (on individual machines as most will have `.aws/Credentials` configured on their local - _TODO_ Use the default chain to include a junit.
* ToDo: Testing

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfiguration.java [hunk 1]
```diff
@@ -121,24 +147,37 @@ public RestHighLevelClient createClient() {
...
-              if (this.insecure) {
-                httpClientBuilder.setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE);
-              }
-              return httpClientBuilder;
-            }
-    );
+    if(awsSigv4) {
+      //if aws signing is enabled we will add AWSRequestSigningApacheInterceptor interceptor,
+      //if not follow regular credentials process
+      LOG.info("{} is set, will sign requests using AWSRequestSigningApacheInterceptor", AWS_SIGV4);
```
Comment (AustinTag) [inline data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfiguration.java:153]: nit: can parse the two credential paths into a function for cleanliness. not a blocker at all

File: data-prepper-plugins/elasticsearch/src/test/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfigurationTests.java [hunk 2]
```diff
@@ -12,26 +12,30 @@
...
+    private final String TEST_DEFAULT_AWS_REGION = "us-east-1";
     private final Integer TEST_CONNECT_TIMEOUT = 5;
     private final Integer TEST_SOCKET_TIMEOUT = 10;
     private final String TEST_CERT_PATH = Objects.requireNonNull(getClass().getClassLoader().getResource("root-ca.pem")).getFile();
 
     @Test
     public void testReadConnectionConfigurationDefault() {
         final PluginSetting pluginSetting = generatePluginSetting(
-                TEST_HOSTS, null, null, null, null, null, false);
+                TEST_HOSTS, null, null, null, null, false, null, null, false);
```
Comment (AustinTag) [inline data-prepper-plugins/elasticsearch/src/test/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfigurationTests.java:31]: I don't see a test that passes in an AWS region

**Review by AustinTag — COMMENTED** [review]
Just think there is a small gap in the unit testing for ConnectionConfiguration. I'm fine with approving, assuming that this has been manually tested. Testing with an integ test with github actions will be hard from a security standpoint. It will likely need to happen in a pipeline further down the road.

File: data-prepper-plugins/elasticsearch/src/test/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfigurationTests.java (hunk 2 above)
Comment (kowshikn) [reply data-prepper-plugins/elasticsearch/src/test/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfigurationTests.java:31]: @yadavcbala  has a default which uses "us-east-1", I am changing this in my PR with null and making sure users pass a region.

**Comment by kowshikn** [comment]
Closing this PR for now, raised PR against the v0.7.x branch for patch release.

**Pull request closed without merging**